core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java

```diff
@@ -287,6 +294,69 @@ public List<Namespace> listNamespaces(Namespace namespace) throws NoSuchNamespac
   public void close() throws IOException {
     namespaces.clear();
     tables.clear();
+    views.clear();
+  }
+
+  @Override
+  public List<TableIdentifier> listViews(Namespace namespace) {
+    if (!namespaceExists(namespace) && !namespace.isEmpty()) {
+      throw new NoSuchNamespaceException(
+          "Cannot list views for namespace. Namespace does not exist: %s", namespace);
+    }
+
+    return views.keySet().stream()
+        .filter(t -> namespace.isEmpty() || t.namespace().equals(namespace))
+        .sorted(Comparator.comparing(TableIdentifier::toString))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public View loadView(TableIdentifier identifier) {
+    if (isValidIdentifier(identifier) && views.containsKey(identifier)) {
+      return new BaseView(newViewOps(identifier), fullTableName(name(), identifier));
+    }
+
+    throw new NoSuchViewException("Invalid view identifier: %s", identifier);
+  }
+
+  @Override
+  protected InMemoryViewOperations newViewOps(TableIdentifier identifier) {
+    return new InMemoryViewOperations(io, identifier);
+  }
+
+  @Override
+  public boolean dropView(TableIdentifier identifier) {
+    return null != views.remove(identifier);
+  }
+
+  @Override
+  public void renameView(TableIdentifier from, TableIdentifier to) {
```

Comment:
Looks like this isn't thread-safe. Do we note that anywhere in Javadoc for this class? Seems likely that the table methods are also not thread-safe.

Reply:
yes this isn't thread-safe (same applies for `renameTable()`) and we currently don't have Javadoc or anything that would indicate this. I'll make a separate PR and will follow-up on this